src/test/compile-fail/self_type_keyword.rs

```diff
@@ -20,14 +20,18 @@ pub fn main() {
     match 15 {
         ref Self => (),
         //~^ ERROR expected identifier, found keyword `Self`
+        //~^^ ERROR match bindings cannot shadow unit structs
```

Comment:
(optional) Could we avoid these extra errors by moving `struct (from earlier in the file) into a sub-module?